auth.go

```diff
@@ -1,102 +1,111 @@
-func (c *AuthCircuit) PrepareInputs(in TypedInputs) (map[string]interface{}, error) {
+// AuthPubSignals auth.circom public signals
+type AuthPubSignals struct {
+	Challenge *big.Int         `json:"challenge"`
+	UserState *merkletree.Hash `json:"userState"`
+	UserID    *core.ID         `json:"userID"`
+}
 
-	authInputs, ok := in.(AuthInputs)
-	if !ok {
-		return nil, errors.New("wrong type of input arguments")
-	}
-	inputs := make(map[string]interface{})
-
-	inputs["userID"] = authInputs.ID.BigInt().String()
-	inputs["userState"] = authInputs.State.StateStr()
-
-	inputs["userAuthClaim"] = bigIntArrayToStringArray(authInputs.AuthClaim.Slots)
-
-	inputs["userAuthClaimMtp"] = bigIntArrayToStringArray(PrepareSiblings(authInputs.AuthClaim.Proof.Siblings, AuthenticationLevels))
-	inputs["userAuthClaimNonRevMtp"] = bigIntArrayToStringArray(PrepareSiblings(authInputs.AuthClaimNonRevocationProof.Siblings, AuthenticationLevels))
-
-	inputs["userClaimsTreeRoot"] = authInputs.State.ClaimsRootStr()
-	inputs["userRevTreeRoot"] = authInputs.State.RevocationRootStr()
-	inputs["userRootsTreeRoot"] = authInputs.State.RootOfRootsRootStr()
-
-	inputs["challenge"] = authInputs.Challenge.String()
-	inputs["challengeSignatureR8x"] = authInputs.Signature.R8.X.String()
-	inputs["challengeSignatureR8y"] = authInputs.Signature.R8.Y.String()
-	inputs["challengeSignatureS"] = authInputs.Signature.S.String()
-
-	if authInputs.AuthClaimNonRevocationProof.NodeAux == nil {
-		inputs["userAuthClaimNonRevMtpAuxHi"] = merkletree.HashZero.BigInt().String()
-		inputs["userAuthClaimNonRevMtpAuxHv"] = merkletree.HashZero.BigInt().String()
-		inputs["userAuthClaimNonRevMtpNoAux"] = new(big.Int).SetInt64(1).String() // (yes it's isOld = 1)
-	} else {
-		inputs["userAuthClaimNonRevMtpNoAux"] = new(big.Int).SetInt64(0).String() // (no it's isOld = 0)
-		if authInputs.AuthClaimNonRevocationProof.NodeAux.HIndex == nil {
-			inputs["userAuthClaimNonRevMtpAuxHi"] = merkletree.HashZero.BigInt().String()
-		} else {
-			inputs["userAuthClaimNonRevMtpAuxHi"] = authInputs.AuthClaimNonRevocationProof.NodeAux.HIndex.BigInt().String()
-		}
-		if authInputs.AuthClaimNonRevocationProof.NodeAux.HValue == nil {
-			inputs["userAuthClaimNonRevMtpAuxHv"] = merkletree.HashZero.BigInt().String()
-		} else {
-			inputs["userAuthClaimNonRevMtpAuxHv"] = authInputs.AuthClaimNonRevocationProof.NodeAux.HValue.BigInt().String()
-		}
+// CircuitOutputUnmarshal unmarshal auth.circom public inputs to AuthPubSignals
+func (a *AuthPubSignals) CircuitOutputUnmarshal(data []byte) error {
+	var sVals []string
+	err := json.Unmarshal(data, &sVals)
+	if err != nil {
+		return err
 	}
 
-	return inputs, nil
-}
+	if len(sVals) != 3 {
+		return fmt.Errorf("invalid number of Output values expected {%d} got {%d} ", 3, len(sVals))
+	}
 
-// AuthInputs ZK inputs
-type AuthInputs struct {
-	ID *core.ID
+	var ok bool
+	if a.Challenge, ok = big.NewInt(0).SetString(sVals[0], 10); !ok {
```

Comment:
Would be good to save these values not to fields of `a`, but to temporary variables, and in the end if there are no errors change the fields finally.